tests/test_core_queries.py

```
@@ -191,3 +192,53 @@ def test_create_taxons_with_wrong_annotation():
     with pytest.raises(ValidationError):
         data["annotations"][0]["annotationtypecv"] = "A wrong CV term"
         schemas.TaxonomicClassifierCreate(**data)
+
+
+@pytest.fixture(scope="function")
+def timeseries_base_data():
+    return {
+        "resultid": 1,
+        "datavalue": 1,
+        "valuedatetime": datetime(2019, 10, 23, 8, 0),
+        "valuedatetimeutcoffset": 0,
+        "censorcodecv": "Not censored",
+        "qualitycodecv": "None",
+        "timeaggregationinterval": 100,
+        "timeaggregationintervalunitsid": 16,
+    }
+
+
+async def upsert_tsrv_table(db_conn, data):
+    records = await insert_timeseries_resultvalues(db_conn, schemas.TimeSeriesResultValuesCreate(**data))
+    return dict(records)
+
+
+@pytest.mark.docker
+@pytest.mark.asyncio
+async def test_insert_timeseries_resultvalues(db_conn, timeseries_base_data):
+    actual_row = await upsert_tsrv_table(db_conn, timeseries_base_data)
+    for k in timeseries_base_data.keys():
+        assert timeseries_base_data[k] == actual_row[k]
+
+
+@pytest.mark.docker
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "new_key, new_value, expected",
+    [
+        ("datavalue", 2, 1),  # We don't want to update datavalue
+        ("qualitycodecv", "Good", "Good"),
+        ("timeaggregationinterval", 1, 1),
+    ],
+)
+async def test_upsert_timeseries_resultvalues(db_conn, timeseries_base_data, new_key, new_value, expected):
+    base_row = await upsert_tsrv_table(db_conn, timeseries_base_data)
+    timeseries_base_data[new_key] = new_value
+    actual_row = await upsert_tsrv_table(db_conn, timeseries_base_data)
+
+    assert actual_row[new_key] == expected
+    assert base_row["valueid"] == actual_row["valueid"]
+
+    for k in base_row.keys():
+        if k != new_key:
```

so, the behaviour is that datavalue is never updated, but silently ignored in case of duplicates (?)

Suggest we look at a way to throw errors if it happens that a) we already have a value  and b) the value differs. But, I don't think this blocks a merge and is only a future improvement.

---

The idea is that it never happens , but you are right, the code should throw error if/when it happens. :-)

---

Yeap, guess this can give various unintended effects that we need to think of in the future.